connect/clients/ethereum.py

```diff
@@ -0,0 +1,321 @@
+"""
+ethereum.py
+
+Ethereum client that provides functionality to interact with a configurable blockchain network.
+Service instance is bound to data attributes and accessed through "get" functions.
+"""
+import asyncio
+import logging
+import time
+from web3 import Web3
+from eth_account import Account
+from hexbytes import HexBytes
+from connect.config import get_settings
+from connect.exceptions import EthereumNetworkConnectionError
+from threading import Thread
+from typing import Callable, List, Optional
+
+
+logger = logging.getLogger(__name__)
+# client instances
+eth_client = None
+
+# ******************************************
+# Ethereum client services
+# ******************************************
+class EthereumClient:
+    """
+    Wrapper for LFH that utilizes the Web3 library for interacting with an
+    Ethereum blockchain network
+    """
+
+    def __init__(self, configs):
+        self._connected = False
+        self._client_connection = None
+        self._eth_network_uri = configs['eth_network_uri']
+        self._default_account = None
+
+        self._client_connection = Web3(Web3.HTTPProvider(self._eth_network_uri))
```

Comment:
To the point above. I tested a bit and confirmed that the Web3 client doesn't raise an exception when initialized. In the snippet below I don't have an eth node running or access to one. Additional the URL can be None and no exception is raised.
```
>>> from web3 import Web3
>>> client =  Web3(Web3.HTTPProvider("http://127.0.0.1:8545"))
>>> client.isConnected()
False

>>> client =  Web3(Web3.HTTPProvider(None))
>>> client.isConnected()
False
```
With that in mind I would recommend opting for a single `self._client_connection` or `self._client` attribute and just check the `isConnected()` method.